zhaquirks/xiaomi/aqara/opple_switch.py

```diff
@@ -0,0 +1,417 @@
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+
+from zhaquirks.const import (
+    ARGS,
+    ATTR_ID,
+    BUTTON,
+    BUTTON_1,
+    BUTTON_2,
+    CLUSTER_ID,
+    COMMAND_BUTTON_DOUBLE,
+    COMMAND_BUTTON_HOLD,
+    COMMAND_BUTTON_SINGLE,
+    COMMAND_DOUBLE,
+    COMMAND_SINGLE,
+    DEVICE_TYPE,
+    ENDPOINT_ID,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    OUTPUT_CLUSTERS,
+    PRESS_TYPE,
+    PROFILE_ID,
+    VALUE,
+    ZHA_SEND_EVENT,
+)
+from zhaquirks.xiaomi import (
+    BasicCluster,
+    DeviceTemperatureCluster,
+    OnOffCluster,
+    XiaomiCustomDevice,
+    XiaomiMeteringCluster,
+)
+
+from .opple_remote import MultistateInputCluster, OppleCluster
+
+BOTH_BUTTONS = "both_buttons"
+PRESS_TYPES = {0: "hold", 1: "single", 2: "double", 3: "triple", 255: "release"}
+
+
+class OppleOperationMode(t.uint8_t, Enum):
+    """Opple operation_mode enum."""
+
+    Decoupled = 0x00
+    Coupled = 0x01
+
+
+class OppleSwitchMode(t.uint8_t, Enum):
+    """Opple switch_mode enum."""
+
+    Fast = 0x01
+    Multi = 0x02
+
+
+class OppleSwitchType(t.uint8_t, Enum):
+    """Opple switch_type enum."""
+
+    Toggle = 0x01
+    Momentary = 0x02
+
+
+class OppleIndicatorLight(t.uint8_t, Enum):
+    """Opple indicator light enum."""
+
+    Normal = 0x00
+    Reverse = 0x01
+
+
+class OppleSwitchCluster(OppleCluster):
+    """Xiaomi mfg cluster implementation."""
+
+    if hasattr(OppleCluster, "attributes"):
+        attributes = copy.deepcopy(OppleCluster.attributes)
+    elif hasattr(OppleCluster, "manufacturer_attributes"):
+        attributes = copy.deepcopy(OppleCluster.manufacturer_attributes)
+    else:
+        attributes = {}
+
+    attributes.update(
+        {
+            0x0002: ("power_outage_count", t.uint8_t),
+            0x000A: ("switch_type", OppleSwitchType),
+            0x00F0: ("reverse_indicator_light", OppleIndicatorLight),
+            0x0125: ("switch_mode", OppleSwitchMode),
+            0x0200: ("operation_mode", OppleOperationMode),
+            0x0201: ("power_outage_memory", t.Bool),
+            0x0202: ("auto_off", t.Bool),
+            0x0203: ("do_not_disturb", t.Bool),
+        }
+    )
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == 0x00FC:
+            self._current_state = PRESS_TYPES.get(value)
+            event_args = {
+                BUTTON: self.endpoint.endpoint_id,
+                PRESS_TYPE: self._current_state,
+                ATTR_ID: attrid,
+                VALUE: value,
+            }
+            action = "{}_{}".format(self.endpoint.endpoint_id, self._current_state)
+            self.listener_event(ZHA_SEND_EVENT, action, event_args)
+            # show something in the sensor in HA
+            super()._update_attribute(0, action)
+
+
+class XiaomiOpple2ButtonSwitchBase(XiaomiCustomDevice):
+    """Xiaomi Opple 2 Button Switch."""
+
+    manufacturer_id_override = 0x115F
```

Comment:
If we define the attributes as manufacturer specific I think we can remove these